ext/bcmath/libbcmath/src/div.c

```diff
@@ -31,219 +31,436 @@
 
 #include "bcmath.h"
 #include "private.h"
+#include "convert.h"
 #include <stdbool.h>
 #include <stddef.h>
 #include <string.h>
 #include "zend_alloc.h"
 
+static const BC_VECTOR POW_10_LUT[9] = {
+	1, 10, 100, 1000, 10000, 100000, 1000000, 10000000, 100000000
+};
 
-/* Some utility routines for the divide:  First a one digit multiply.
-   NUM (with SIZE digits) is multiplied by DIGIT and the result is
-   placed into RESULT.  It is written so that NUM and RESULT can be
-   the same pointers.  */
-
-static void _one_mult(unsigned char *num, size_t size, int digit, unsigned char *result)
+/*
+ * Use this function when the number of array elements in divisor is 1, that is, when divisor is not divided into multiple array elements.
+ * The algorithm can be simplified if divisor is not divided, and this function is an optimization of bc_standard_div.
+ */
+static inline void bc_fast_div(
+	BC_VECTOR *numerator_vector, size_t numerator_arr_size, BC_VECTOR divisor_vector, BC_VECTOR *quot_vector, size_t quot_arr_size)
 {
-	size_t carry, value;
-	unsigned char *nptr, *rptr;
+	size_t numerator_top_index = numerator_arr_size - 1;
+	size_t quot_top_index = quot_arr_size - 1;
+	for (size_t i = 0; i < quot_arr_size - 1; i++) {
+		if (numerator_vector[numerator_top_index - i] < divisor_vector) {
+			quot_vector[quot_top_index - i] = 0;
+			/* numerator_vector[numerator_top_index - i] is always a smaller number than divisor_vector. Therefore, there will be no overflow. */
```

Comment:
I don't understand why the numerator is smaller than the divisor?

Reply:
This is because the process is inside `if (numerator_vector[numerator_top_index - i] < divisor_vector) {`.
It is certainly difficult to understand when written as "always", so I will change the wording.

Reply:
Fixed
https://github.com/php/php-src/pull/14660/commits/28bf55df6c70b99d179e18bddeaaffd41d077526